src/Javis.jl

```diff
@@ -317,6 +329,91 @@ function get_javis_frame(video, objects, frame)
         # if object is in global layer this changes the background settings
         update_background_settings!(background_settings, object)
     end
+end
+
+function get_layer_frame(video, layer, frame)
+    Drawing(layer.width, layer.height, :image)
+    objects = layer.children
+    render_objects(objects, video, frame)
+    if frame in get_frames(layer)
+        # call currently active actions and their transformations for each layer
+        actions = layer.actions
+        if !isempty(actions)
+            for action in actions
+                #actions for layers are handled in a different way
+                # we don't calculate the relative frames for that
+                # can find out a way to pre_compute frames for actions applie to layers
+                # this is not a final fix..
+                if frame in get_frames(action)
+                    action.func(video, layer, action, frame)
+                elseif frame > last(get_frames(action)) && action.keep
+                    # call the action on the last frame i.e. disappeared things stay disappeared
+                    action.func(video, layer, action, last(get_frames(action)))
+                end
+            end
+        end
+    end
+    img_layer = image_as_matrix()
+    finish()
+    return img_layer
+end
+
+function apply_layer_actions(video, layers, frame)
```

Comment:
naming is a bit complicated as it seems like there are actions in layer.actions and things that are controlled by the layer settings?